lib/teem.rb

```diff
@@ -0,0 +1,179 @@
+module Teem
+  require 'net/http'
+  require 'json'
+
+  CLIENT_ID=ENV["TEEM_CLIENT_ID"]
+  CLIENT_SECRET=ENV["TEEM_CLIENT_SECRET"]
+  REDIRECT_URI=ENV["TEEM_REDIRECT_URI"]
+
+  USERNAME=ENV["TEEM_USERNAME"]
+  PASSWORD=ENV["TEEM_PASSWORD"]
+
+  ORG_NAME=ENV["TEEM_ORG_NAME"]
+
+  def self.request(methods, access_token, url, json_obj = nil)
+    uri = URI.parse(url)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+    if methods == :get
+      request = Net::HTTP::Get.new(uri.request_uri)
+    elsif methods == :post
+      request = Net::HTTP::Post.new(uri.request_uri)
+    elsif methods == :patch
+      request = Net::HTTP::Patch.new(uri.path)
+    elsif methods == :delete
+      request = Net::HTTP::Delete.new(uri.request_uri)
+    end
+
+    if access_token
+      request.add_field("Authorization", "Bearer #{access_token}")
+    end
+
+    if json_obj
+      request.add_field('Content-Type', 'application/json')
+      request.body = json_obj
+    end
+
+    response = http.request(request)
+
+    return response.body ? JSON.parse(response.body) : nil
+  end
+
+  # Authorizes against the Teem API. Required before any API actions.
+  # Returns the Teem API JSON object containing the access token.
+  def self.teem_authorize
```

Comment:
Let's rename all the functions in the 'Teem' module to not have the 'teem_' prefix. It makes them redundant to call, e.g. Teem.teem_get_users() should be Teem.get_users().